src/Catty/SavePlotSVGBrick+Instructions.swift

```diff
@@ -0,0 +1,97 @@
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see http://www.gnu.org/licenses/.
+ */
+
+extension SavePlotSVGBrick: CBInstructionProtocol {
+
+    func instruction() -> CBInstruction {
+        .action { context in SKAction.run(self.actionBlock(context: context)) }
+    }
+
+    func actionBlock(context: CBScriptContextProtocol) -> () -> Void {
+        guard let object = self.script?.object,
+            let spriteNode = object.spriteNode
+            else { fatalError("This should never happen!") }
+
+        return {
+            var filename = context.formulaInterpreter.interpretString(self.filename!, for: object)
+            if let number = Double(filename) {
+                filename = number.displayString
+            }
+            var paths = ""
+            for i in 0..<spriteNode.penConfiguration.previousCutPositionLines.count
+            {
+                paths = paths + self.getLinePath(with: spriteNode.penConfiguration.previousCutPositionLines[i]!) + "\n"
+            }
+            paths = paths + self.getLinePath(with: spriteNode.penConfiguration.previousCutPositions)
+        
+            self.saveSVGPlot(with: paths, to: filename, width: Int(object.scene.width() ?? "0") ?? 0, height: Int(object.scene.height() ?? "0") ?? 0)
+        }
+    }
+    
+    private func getDocumentsDirectory() -> URL {
+        let paths = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask)
+        return paths[0]
+    }
+    
+    private func saveSVGPlot(with paths: String, to filename:String, width : Int, height : Int){
+        var filecontent = "<?xml version=\"1.0\" standalone=\"no\"?>\n<!DOCTYPE svg PUBLIC \"-//W3C//DTD SVG 1.1//EN\" \"http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd\">\n"
+        filecontent += "<svg width=\"" + String(width) + "\" height=\"" + String(height) + "\" viewBox=\"0 0 " + String(width) + " " + String(height) + "\" style=\"background-color:#ffffff\" xmlns=\"http://www.w3.org/2000/svg\" version=\"1.1\">\n"
+        filecontent += "<title>Plotter export</title>\n"
+        var filename = filename
+        
+        filecontent += paths
+        filecontent += "\n</svg>"
+        
+        
+        if(!filename.hasSuffix(".svg")){
+            filename = filename + ".svg"
+        }
+        let file = self.getDocumentsDirectory().appendingPathComponent(filename)
+        let debugfile = self.getDocumentsDirectory().appendingPathComponent("filename.txt")
+        do {
+            try filecontent.write(to: file, atomically: true, encoding: String.Encoding.utf8)
+            try filecontent.write(to: debugfile, atomically: true, encoding: String.Encoding.utf8)
+        } catch {
+            // TODO: failed to write file
```

Comment:
ist kein error handling gewünscht? Wenn keines gewünscht ist kannst du auch try? schreiben dann brauchst kein do catch

Reply:
Derweil is noch keines Gewünscht :)